### PR TITLE
Adds signed in user to backend v2 plugins requests

### LINKED ISSUE #12043: Backend Plugins: Pass user information
It is useful to restrict permission in backend plugin.
For example, I want to use the user information to restrict log viewing permission in sumologic plugin.
https://github.com/mtanda/grafana-sumologic-datasource

I guess, we need to extend backend plugin protocol, and pass it like following code.
https://github.com/mtanda/grafana/commit/512ecda60918a55d418de182fa93845cd67a33c1

@bergquist This is the feature request what I told you in GrafanaCon.

### PR DESCRIPTION
**Pull request by bergquist**
closes #12043

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,7 +31,8 @@ require (
 	github.com/gorilla/websocket v1.4.1
 	github.com/gosimple/slug v1.4.2
 	github.com/grafana/grafana-plugin-model v0.0.0-20190930120109-1fc953a61fb4
-	github.com/grafana/grafana-plugin-sdk-go v0.16.0
+	github.com/grafana/grafana-plugin-sdk-go v0.17.1-0.20200304132035-b6b43f60bc5b
```
Comment (bergquist) [inline go.mod]: This should be updated to point at a tag.

**Comment by bergquist** [comment]
This PR became quite messy :/

**Comment by marefr** [comment]
Just merged #22578. Please merge in latest master, resolve conflicts and then run `go mod tidy` followed by `go mod vendor`.

File: go.mod (hunk 1 above)
Comment (marefr) [reply go.mod]: We probably need to document how to work with modules and using a non tagged version of SDK.

I usually add a replace directive in go.mod:
`replace github.com/grafana/grafana-plugin-sdk-go => github.com/grafana/grafana-plugin-sdk-go <remote branch name>`
Then run `GO111MODULE=on go mod tidy` followed by `GO111MODULE=on go mod vendor`.

go.mod replace `<remote branch name>` should then automatically change to reflect head commit in that branch. 

This way the module version shouldn't need to change in require section.

**Review by marefr — APPROVED** [review]
LGTM, but should merge dependent PR in SDK and tag a new version and then update this PR before merge.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Pull request merged**